lib/harvestdor/indexer.rb

```diff
@@ -63,11 +63,16 @@ def harvest_and_index(each_options = { in_threads: 4 })
     end
 
     def resources
-      druids.map do |x|
-        Harvestdor::Indexer::Resource.new(self, x)
-      end.map do |x|
-        [x, (x.items if x.collection?)]
-      end.flatten.uniq.compact
+      return to_enum(:resources) unless block_given?
+
+      druids.each do |x|
+        resource = Harvestdor::Indexer::Resource.new(self, x)
+        yield resource
+
+        resource.items.each do |member|
+          yield member
+        end if resource.collection?
+      end
```

Comment:
please pardon my ignorance, but it's not obvious to me what this method is doing.  (now, or before).  Please sir, may I have a comment or two?   Is it returning an enumerable object that is essentially a collection of resources?  and it's getting single items in "druids" array and is getting all the items in a collection for collection druids?